File: android/app/src/main/kotlin/com/cleancity/vibe/vibesweep/DeviceCapabilityPlugin.kt
package com.cleancity.vibe.vibesweep

import android.content.Context
import android.os.Build
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class DeviceCapabilityPlugin: FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var context: Context

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "vibesweep/device_capabilities")
        channel.setMethodCallHandler(this)
        context = flutterPluginBinding.applicationContext
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "checkARCoreSupport" -> {
                result.success(checkARCoreSupport())
            }
            "hasHighPerformanceGPU" -> {
                result.success(hasHighPerformanceGPU())
            }
            "getAvailableMemory" -> {
                result.success(getAvailableMemory())
            }
            "getDeviceInfo" -> {
                result.success(getDeviceInfo())
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    private fun checkARCoreSupport(): Boolean {
        // Basic heuristic for ARCore support
        // In production, you would use proper ARCore availability check
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && 
               !isLowEndDevice()
    }

    private fun hasHighPerformanceGPU(): Boolean {
        // Heuristic based on device characteristics
        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as android.app.ActivityManager
        val memoryInfo = android.app.ActivityManager.MemoryInfo()
        activityManager.getMemoryInfo(memoryInfo)
        
        // Consider high-performance if device has more than 4GB RAM
        return memoryInfo.totalMem > 4L * 1024 * 1024 * 1024
    }

    private fun getAvailableMemory(): Int {
        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as android.app.ActivityManager
        val memoryInfo = android.app.ActivityManager.MemoryInfo()
        activityManager.getMemoryInfo(memoryInfo)
        
        // Return available memory in MB
        return (memoryInfo.availMem / (1024 * 1024)).toInt()
    }

    private fun isLowEndDevice(): Boolean {
        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as android.app.ActivityManager
        return activityManager.isLowRamDevice
    }

    private fun getDeviceInfo(): Map<String, Any> {
        return mapOf(
            "manufacturer" to Build.MANUFACTURER,
            "model" to Build.MODEL,
            "androidVersion" to Build.VERSION.RELEASE,
            "sdkVersion" to Build.VERSION.SDK_INT,
            "isLowRamDevice" to isLowEndDevice()
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Runner/DeviceCapabilityPlugin.swift
import Flutter
import UIKit
import ARKit

public class DeviceCapabilityPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "vibesweep/device_capabilities", binaryMessenger: registrar.messenger())
        let instance = DeviceCapabilityPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "checkARKitSupport":
            result(checkARKitSupport())
        case "hasHighPerformanceGPU":
            result(hasHighPerformanceGPU())
        case "getAvailableMemory":
            result(getAvailableMemory())
        case "getDeviceInfo":
            result(getDeviceInfo())
        default:
            result(FlutterMethodNotImplemented)
        }
    }

    private func checkARKitSupport() -> Bool {
        if #available(iOS 11.0, *) {
            return ARWorldTrackingConfiguration.isSupported
        }
        return false
    }

    private func hasHighPerformanceGPU() -> Bool {
        // Heuristic based on device model
        let deviceModel = UIDevice.current.model
        let systemVersion = UIDevice.current.systemVersion
        
        // Consider high-performance for newer devices
        if #available(iOS 12.0, *) {
            return ProcessInfo.processInfo.physicalMemory > 3 * 1024 * 1024 * 1024 // 3GB+
        }
        return false
    }

    private func getAvailableMemory() -> Int {
        let physicalMemory = ProcessInfo.processInfo.physicalMemory
        // Return available memory in MB (simplified calculation)
        return Int(physicalMemory / (1024 * 1024))
    }

    private func getDeviceInfo() -> [String: Any] {
        return [
            "model": UIDevice.current.model,
            "systemName": UIDevice.current.systemName,
            "systemVersion": UIDevice.current.systemVersion,
            "identifierForVendor": UIDevice.current.identifierForVendor?.uuidString ?? "unknown",
            "physicalMemory": ProcessInfo.processInfo.physicalMemory
        ]
    }
}